### PR TITLE
upgrade to 1.10

### PR DESCRIPTION
**Pull request by yuvalif**

### CODE REVIEW

**Comment by fabiand** [comment]
Awesome!

Please add a commit message.

**Comment by yuvalif** [comment]
the commit message is "upgrade to 1.10". should I add more info in it?

File: Gopkg.toml [hunk 1]
```diff
@@ -29,8 +29,11 @@
   branch = "master"
   name = "golang.org/x/net"
 
-
 [[constraint]]
   branch = "master"
   name = "github.com/kubevirt/device-plugin-manager"
 
+[[constraint]]
+  version = "1.10"
```
Comment (yuvalif) [inline Gopkg.toml]: fix it to 1.10

File: Makefile [hunk 2]
```diff
@@ -25,18 +27,13 @@ docker-build: $(patsubst %, docker-build-%, $(DOCKERFILES))
 
 docker-build-%:
 	@cp ${GOPATH}/bin/${notdir $(subst -,/,$*)} ./cmd/$(subst -,/,$*)
-	docker build -t quay.io/kubevirt/device-plugin-$*:latest ./cmd/$(subst -,/,$*)
+	docker build -t ${REGISTRY}/device-plugin-$*:latest ./cmd/$(subst -,/,$*)
```
Comment (yuvalif) [inline Makefile]: allow registry to be set by user

File: Makefile [hunk 3]
```diff
@@ -25,18 +27,13 @@ docker-build: $(patsubst %, docker-build-%, $(DOCKERFILES))
...
 
 docker-push: $(patsubst %, docker-push-%, $(DOCKERFILES))
 
 docker-push-%:
-	docker push quay.io/kubevirt/device-plugin-$*:latest
-
-docker-local-push: $(patsubst %, docker-local-push-%, $(DOCKERFILES))
```
Comment (yuvalif) [inline Makefile]: no need for explicit local registry

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+*
```
Comment (phoracek) [inline .gitignore:1]: why did you add this?

File: Gopkg.toml [hunk 5]
```diff
@@ -28,3 +28,12 @@
 [[constraint]]
   branch = "master"
   name = "golang.org/x/net"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/kubevirt/device-plugin-manager"
```
Comment (phoracek) [inline Gopkg.toml:34]: please fix it to latest commit

File: Makefile [hunk 6]
```diff
@@ -13,7 +15,7 @@ DOCKERFILES = $(sort \
 build: $(patsubst %, build-%, $(PLUGINS))
 
 build-%:
-	cd cmd/$(subst -,/,$*) && go build
```
Comment (phoracek) [inline Makefile:16]: why did you change it to install?

File: Makefile [hunk 7]
```diff
@@ -24,11 +26,21 @@ test-%:
...
 docker-build-%:
-	docker build -t quay.io/kubevirt/device-plugin-$*:latest cmd/$(subst -,/,$*)
+	@cp ${GOPATH}/bin/${notdir $(subst -,/,$*)} ./cmd/$(subst -,/,$*)
+	docker build -t ${REGISTRY}/device-plugin-$*:latest ./cmd/$(subst -,/,$*)
 
 docker-push: $(patsubst %, docker-push-%, $(DOCKERFILES))
 
 docker-push-%:
-	docker push quay.io/kubevirt/device-plugin-$*:latest
+	docker tag ${REGISTRY}/device-plugin-$*:latest ${REGISTRY}/device-plugin-$*
```
Comment (phoracek) [inline Makefile]: is it needed to tag it? is not it tagged by the name automatically?

File: pkg/kvm/kvm.go [hunk 8]
```diff
@@ -88,7 +88,23 @@ func (dpi *KVMDevicePlugin) Allocate(ctx context.Context, r *pluginapi.AllocateR
 	dev.HostPath = "/dev/kvm"
 	dev.ContainerPath = "/dev/kvm"
 	dev.Permissions = "rw"
-	response.Devices = append(response.Devices, dev)
+	var devices []*pluginapi.DeviceSpec
+	devices = append(devices, dev)
+	response.ContainerResponses = append(response.ContainerResponses, &pluginapi.ContainerAllocateResponse {
+		Devices: devices })
```
Comment (phoracek) [inline pkg/kvm/kvm.go]: Did you run gofmt on this? Just curious.

**Review by phoracek — CHANGES_REQUESTED** [review]
Thanks for the Makefile changes and cleanup. Looks good, just a nit.

File: .gitignore (hunk 4 above)
Comment (yuvalif) [reply .gitignore:1]: when i do "git status" I don't want to see the binaries. the content of the gitignore is to ignore all files that does not have a suffix.
another option is to just list all the binary names here

File: Makefile (hunk 6 above)
Comment (yuvalif) [reply Makefile:16]: today the project is small, so rebuilding from scratch is not a big deal. but if you have large codebase, "go install" make sure that you only compile modified packages

File: Makefile (hunk 7 above)
Comment (yuvalif) [reply Makefile]: not sure... will check

File: pkg/kvm/kvm.go (hunk 8 above)
Comment (yuvalif) [reply pkg/kvm/kvm.go]: no... will add to the makefile :-)

File: Gopkg.toml (hunk 5 above)
Comment (yuvalif) [reply Gopkg.toml:34]: isn't "master" the latest commit?

File: Makefile (hunk 7 above)
Comment (yuvalif) [reply Makefile]: you are right. tag is not needed

File: Makefile (hunk 6 above)
Comment (phoracek) [reply Makefile:16]: did not know that, thank you

File: Gopkg.toml (hunk 5 above)
Comment (phoracek) [reply Gopkg.toml:34]: I mean latest commit right now, set commit ID, not sure if it would be better since we keep vendor dir

File: .gitignore (hunk 4 above)
Comment (phoracek) [reply .gitignore:1]: Are there only binaries for our DPs or is there more because of the switch to go install?

Comment (yuvalif) [reply .gitignore:1]: the switch to "go install" actually keeps the binaries out of the git area. but when we want to build the docker images they are copied back in.
adding .gitignore was needed regardless of the change in the build

File: Gopkg.toml (hunk 5 above)
Comment (yuvalif) [reply Gopkg.toml:34]: do you want me to change the dependency to branch "release-1.9"? this will force k8s of 1.9 as well?
don't think it makes sense to point to latest commit. maybe create a 1.10 brancg on the dpm, and will point there?

Comment (phoracek) [reply Gopkg.toml:34]: Thinking again about it, it's not a big deal with vendoring and lock file.

**Review by phoracek — APPROVED** [review]
